Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -946,6 +970,49 @@ open class ComposerVC: _ViewController,
         suggestionsVC.view.removeFromSuperview()
     }
 
+    /// The links in the current input text have changed
+    /// - Parameter links: The new parsed links from the input text.
+    open func didChangeLinks(_ links: [TextLink]) {
+        /// We only try to display the link preview of the first link.
+        guard let link = links.first else {
+            dismissLinkPreview()
+            return
+        }
+
+        enrichUrlDebouncer.execute { [weak self] in
+            self?.channelController?.client.enrichUrl(link.url) { [weak self] result in
```

Comment:
This is also an argument to use the `channelController` for enriching urls, instead of the `client`.

Reply:
Some as above 👍